llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -34,8 +34,11 @@ void VPlanTransforms::VPInstructionsToVPRecipes(
     ScalarEvolution &SE, const TargetLibraryInfo &TLI) {
 
   ReversePostOrderTraversal<VPBlockDeepTraversalWrapper<VPBlockBase *>> RPOT(
-      Plan->getEntry());
+      Plan->getVectorLoopRegion());
   for (VPBasicBlock *VPBB : VPBlockUtils::blocksOnly<VPBasicBlock>(RPOT)) {
+    // Skip blocks outside region
```

Comment:
Unfortunately yes. The way GraphTraits work, I don't think theres' a way to avoid 'escaping' as we can't access the starting node when constructing the iterators for each visited node :(